src/DocChecks.jl

```diff
@@ -366,6 +378,34 @@ function print_indented(buffer::IO, str::AbstractString; indent = 4)
     end
 end
 
+function fix_doctest(result::Result, str, doc::Documents.Document)
+    code = result.block.code
+    open(Base.find_source_file(result.file), "r+") do f
+        content = read(f, String)
+        seekstart(f)
+        truncate(f, 0)
+        # first look for the entire code block
+        codeidx = findnext(code, content, 1)
+        @assert codeidx != 0:-1 # this should never happen
```

Comment:
If you have CRLF line endings then this throws. It also leaves the source file empty on error right now.. so even if we fix this particular case, I think it would probably be a good idea to `try ... catch` all of this and restore the original file when are any errors (and then rethrow).

Reply:
Good catch. We could potentially try to detect the line ending style, unless there could sometimes be a mix of the two?
Probably good to wrap in try catch anyway